v2/internal/reflecthelpers/reflect_helpers.go

```diff
@@ -379,3 +379,47 @@ func setPropertyCore(obj any, propertyPath []string, value any) (err error) {
 	field.Set(reflect.ValueOf(value))
 	return nil
 }
+
+// PopulateStruct is far from complete/perfect and shouldn't be used in production code.
+// Test usage is OK.
+// deprecated: Don't use it in production code.
+func PopulateStruct(s any) {
```

Comment:
I don't follow what this is for.

Reply:
It's for testing, see the tests at v2/api/apimanagement/customizations/subscription_extensions_test.go for example.

Specifically what I wanted to test was: 

- Fill a given `XyzSecrets` object. For example for `apimanagement` above, the `SubscriptionOperatorSecrets` object has a `primaryKey` and `secondaryKey` field.
- Call the `secretsSpecified` helper on the above object, which returns the set of secrets requested.
- Ensure that the set of secret keys matches the set of JSON keys on the objects.

Or maybe put another way, I wanted to assert that this:
```
const (
	primaryKey   = "primaryKey"
	secondaryKey = "secondaryKey"
)
```

Equals the non-$propertyBag keys from this:
```
type SubscriptionOperatorSecrets struct {
	PrimaryKey   *genruntime.SecretDestination `json:"primaryKey,omitempty"`
	PropertyBag  genruntime.PropertyBag        `json:"$propertyBag,omitempty"`
	SecondaryKey *genruntime.SecretDestination `json:"secondaryKey,omitempty"`
}
```

This being in reflecthelpers is awkward though because we use that in production and this method is super hacky. I moved it to a new testcommon/reflect package which should make it more obvious it's just for test.